integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/container/MySQLContainer.java

```diff
@@ -0,0 +1,158 @@
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.StringUtils;
+import org.rnorth.ducttape.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.Network;
+
+public class MySQLContainer extends BaseContainer {
+  public static final Logger LOG = LoggerFactory.getLogger(MySQLContainer.class);
+
+  public static final String DEFAULT_IMAGE = "mysql:8.0";
+  public static final String HOST_NAME = "gravitino-ci-mysql";
+  public static final int MYSQL_PORT = 3306;
+  public static final String USER_NAME = "root";
+  public static final String PASSWORD = "root";
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  protected MySQLContainer(
+      String image,
+      String hostName,
+      Set<Integer> ports,
+      Map<String, String> extraHosts,
+      Map<String, String> filesToMount,
+      Map<String, String> envVars,
+      Optional<Network> network) {
+    super(image, hostName, ports, extraHosts, filesToMount, envVars, network);
+  }
+
+  @Override
+  protected void setupContainer() {
+    super.setupContainer();
+    withLogConsumer(new PrintingContainerLog(format("%-14s| ", "MySQLContainer")));
+  }
+
+  @Override
+  public void start() {
+    super.start();
+    Preconditions.check("MySQL container startup failed!", checkContainerStatus(5));
+  }
+
+  @Override
+  protected boolean checkContainerStatus(int retryLimit) {
+    int nRetry = 0;
+    boolean isMySQLContainerReady = false;
+    int sleepTimeMillis = 20_00;
+    while (nRetry++ < retryLimit) {
+      try {
+        String[] commandAndArgs =
+            new String[] {
+              "mysqladmin",
+              "ping",
+              "-h",
+              "localhost",
+              "-u",
+              getUsername(),
+              String.format("-p%s", getPassword())
+            };
+        Container.ExecResult execResult = executeInContainer(commandAndArgs);
+        if (execResult.getExitCode() != 0) {
+          String message =
+              format(
+                  "Command [%s] exited with %s",
+                  String.join(" ", commandAndArgs), execResult.getExitCode());
+          LOG.error("{}", message);
+          LOG.error("stderr: {}", execResult.getStderr());
+          LOG.error("stdout: {}", execResult.getStdout());
+        } else {
+          LOG.info("MySQL container startup success!");
+          isMySQLContainerReady = true;
+          break;
+        }
+        LOG.info(
+            "MySQL container is not ready, recheck({}/{}) after {}ms",
+            nRetry,
+            retryLimit,
+            sleepTimeMillis);
+        await().atLeast(sleepTimeMillis, TimeUnit.MILLISECONDS);
```

Comment:
@mchades  I've used `await().atLeast(sleepTimeMillis, TimeUnit.MILLISECONDS)` here, isn't that correct ?

Reply:
ok